pkg/clienthandler/rtu_client.go

```diff
@@ -46,6 +46,9 @@ func (mb *rtuSerialTransporter) Send(aduRequest []byte) (aduResponse []byte, err
 	if err != nil {
 		return
 	}
+	if bytesToRead == -1 {
+		bytesToRead = specialCalculateResponseLength(aduRequest, data[3])
+	}
```

Comment:
this code should be close to the line number 39. Logically they should be grouped together.